### PR TITLE
PWN-6053 - not enough funds state added to search

### PR DESCRIPTION
**Pull request by kabramovp2p**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-6053

## Description of Work

Not enough founds state added to search

## Screenshot

Android Emulator - Resizable_API_33:5554 2022-12-05 13-35-30

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/send/ui/search/NewSearchPresenter.kt [hunk 1]
```diff
@@ -75,6 +86,12 @@ class NewSearchPresenter(
         }
     }
 
+    override fun onBuyClicked() {
+        usdcTokenForBuy?.let {
+            view?.showBuyScreen(it)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/send/ui/search/NewSearchPresenter.kt:91]: can be method reference
should we log something if usdc is null?

File: app/src/main/res/values/send_strings.xml [hunk 2]
```diff
@@ -18,8 +18,11 @@
     <string name="search_caution_error">Caution: this address has no funds</string>
     <string name="search_no_other_tokens_description">%s token account</string>
     <string name="search_no_other_tokens_error">You cannot send SOL to this address because it belongs to another token (%s)</string>
+    <string name="search_no_founds_error">Account creation for this address is not possible due to insufficient funds</string>
     <string name="search_yourself_error">You cannot send tokens to yourself</string>
     <string name="search_yourself_description">Your address</string>
+    <string name="search_receive">Receive</string>
```
Comment (gslevinkov) [inline app/src/main/res/values/send_strings.xml:24]: can be moved to common strings

see common_wallet for example

**Review by gslevinkov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/send/interactor/SearchInteractor.kt [hunk 3]
```diff
@@ -94,4 +96,10 @@ class SearchInteractor(
     }
 
     fun isOwnPublicKey(publicKey: String) = publicKey == tokenKeyProvider.publicKey
+
+    private fun Token.Active.hasFundsForSend(relayInfo: RelayInfo): Boolean {
+        // TODO modify logic according to business needs
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/send/interactor/SearchInteractor.kt:101]: task number?

File: app/src/main/java/org/p2p/wallet/send/ui/search/NewSearchPresenter.kt [hunk 4]
```diff
@@ -89,7 +89,7 @@ class NewSearchPresenter(
     override fun onBuyClicked() {
         usdcTokenForBuy?.let {
             view?.showBuyScreen(it)
-        }
+        } ?: Timber.w("Unable to find USDC TokenForBuy!")
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/send/ui/search/NewSearchPresenter.kt:92]: it will not log at all, use exception or Timber.i

**Review by gslevinkov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/send/interactor/SearchInteractor.kt (hunk 3 above)
Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/send/interactor/SearchInteractor.kt:101]: no task number for now (

**Review by 1sakov — APPROVED** [review]

**Pull request merged**